### ISSUE TITLE
netzkino - Unsupported URL

### ISSUE THREAD

**Issue by mouz124** [ytdl-org/youtube-dl#30457]
## Checklist
- [x ] I'm reporting a broken site support
- [x ] I've verified that I'm running youtube-dl version **2021.12.17**
- [x ] I've checked that all provided URLs are alive and playable in a browser
- [x ] I've checked that all URLs and arguments with special characters are properly quoted or escaped
- [x ] I've searched the bugtracker for similar issues including closed ones

## Verbose log

C:\Temp\ffmpeg-4.4.1-essentials_build>youtube-dl -v https://www.netzkino.de/filme/die-blaue-lagune
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['-v', 'https://www.netzkino.de/filme/die-blaue-lagune']
[debug] Encodings: locale cp1252, fs mbcs, out cp850, pref cp1252
[debug] youtube-dl version 2021.12.17
[debug] Python version 3.4.4 (CPython) - Windows-10-10.0.18362
[debug] exe versions: none
[debug] Proxy map: {}
[generic] die-blaue-lagune: Requesting header
WARNING: Falling back on generic information extractor.
[generic] die-blaue-lagune: Downloading webpage
ERROR: Unable to download webpage: IncompleteRead(5394 bytes read, 759 more expected) (caused by IncompleteRead(5394 bytes read, 759 more expected)); please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmpupik7c6w\build\youtube_dl\extractor\common.py", line 634, in _request_webpage
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmpupik7c6w\build\youtube_dl\YoutubeDL.py", line 2288, in urlopen
  File "C:\Python\Python34\lib\urllib\request.py", line 470, in open
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmpupik7c6w\build\youtube_dl\utils.py", line 2625, in http_response
  File "C:\Python\Python34\lib\http\client.py", line 543, in read
  File "C:\Python\Python34\lib\http\client.py", line 629, in _readall_chunked
  File "C:\Python\Python34\lib\http\client.py", line 701, in _safe_read

## Description

The netzkino URL is broken, i try different link... I have try with ID "watch/a0U01000000yIU5" and the name "/filme/die-blaue-lagune"
I Think the extractor do not found netzkino.py

Bug 9226 looks similar only that extractor is correct here 
https://github.com/ytdl-org/youtube-dl/issues/9226
looks similar only that it is correct here

**Comment by dirkf** [comment]
The Netzkino extractor isn't running because it wants `.../#!/...` in the URL. You can provide it but the extractor fails:
``` 
youtube-dl -v -F 'https://www.netzkino.de/#!/filme/die-blaue-lagune'
[debug] System config: ['--prefer-ffmpeg']
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['-v', '-F', 'https://www.netzkino.de/#!/filme/die-blaue-lagune']
[debug] Encodings: locale UTF-8, fs utf-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2021.12.17
[debug] Python version 3.5.2 (CPython) - Linux-4.4.0-210-generic-i686-with-Ubuntu-16.04-xenial
[debug] exe versions: avconv 4.3, avprobe 4.3, ffmpeg 4.3, ffprobe 4.3
[debug] Proxy map: {}
[Netzkino] die-blaue-lagune: Downloading JSON metadata
ERROR: An extractor error has occurred. (caused by KeyError('posts',)); please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; see  https://yt-dl.org/update  on how to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/youtube_dl/extractor/common.py", line 534, in extract
    ie_result = self._real_extract(url)
  File "/usr/lib/python3/dist-packages/youtube_dl/extractor/netzkino.py", line 45, in _real_extract
    p for p in api_info['posts'] if p['slug'] == video_id)
KeyError: 'posts'
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/youtube_dl/extractor/common.py", line 534, in extract
    ie_result = self._real_extract(url)
  File "/usr/lib/python3/dist-packages/youtube_dl/extractor/netzkino.py", line 45, in _real_extract
    p for p in api_info['posts'] if p['slug'] == video_id)
KeyError: 'posts'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/youtube_dl/YoutubeDL.py", line 815, in wrapper
    return func(self, *args, **kwargs)
  File "/usr/lib/python3/dist-packages/youtube_dl/YoutubeDL.py", line 836, in __extract_info
    ie_result = ie.extract(url)
  File "/usr/lib/python3/dist-packages/youtube_dl/extractor/common.py", line 547, in extract
    raise ExtractorError('An extractor error has occurred.', cause=e)
youtube_dl.utils.ExtractorError: An extractor error has occurred. (caused by KeyError('posts',)); please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; see  https://yt-dl.org/update  on how to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.

$ 
```
Checking the site it appears to have been rewritten completely and possibly is serving DRM content that would be unplayable even if someone re-worked the extractor.

The patch at https://github.com/TobiX/youtube-dl/commit/9605fc7fefffeed254ddae7f2f286693560c3494 that was said to have fixed #9226 fails with a 404 fetching the player JS. That is essentially the same as the Netzkino extractor for yt-dlp.

According to a report at yt-dlp, Netzkino has a YT channel, geo-restricted to Germany.

**Comment by chkuendig** [comment]
I don't think all videos use DRM. For example https://www.netzkino.de/watch/a0U01000000yEWf seems to just directly steam from a mp4 file ( https://pmd.netzkino-seite.netzkino.de/Studio_Hamburg_NK/Eichmanns_Ende_-_Liebe_Verrat_und_Tod.mp4 ) which can be downloaded with WGET without even any referrers or other custom headers.
